src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/resource/ActionPlanController.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.resource
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.service.ActionPlanService
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.mapper.ActionPlanDomainMapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.resource.model.ActionPlanResponse
+
+@RestController
+@RequestMapping(value = ["/action-plans"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class ActionPlanController(
+  private val actionPlanService: ActionPlanService,
+  private val actionPlanDomainMapper: ActionPlanDomainMapper,
+) {
+
+  @GetMapping
+  @ResponseStatus(HttpStatus.OK)
+  fun getActionPlan(@PathVariable prisonNumber: String): ActionPlanResponse =
+    actionPlanDomainMapper.fromDomainToModel(actionPlanService.getActionPlan(prisonNumber))
```

Comment:
I find this short syntax really hard to read sometimes .... and this is one of those times! 😢 
All I can see is the mapper as the "primary" thing ...

How about:
```
fun getActionPlan(@PathVariable prisonNumber: String): ActionPlanResponse =
  actionPlanService.getActionPlan(prisonNumber).let {
    actionPlanDomainMapper.fromDomainToModel(it)
  }
```

To me, I _think_ that reads a little better perhaps because the "primary" thing I see is the service call ...

What do you think?

Reply:
Or alternatively:

```
with(actionPlanService.getActionPlan(prisonNumber)) {
   actionPlanDomainMapper.fromDomainToModel(this)
}
```
Any preference?